tests/format/AllSolidityFeatures/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -1033,23 +1033,23 @@ contract Overrides {
             BazQuxBazQuxBazQux,
             Quuuuuuuuuuuuuux
         ) varName;
-    modifier onlyOwner() virtual {
+    modifier onlyOwner virtual {
```

Comment:
I had no idea you could omit the parentheses in the modifier _definition_. I think we should keep them in the definitions but erase them in the invocations. @frangio what do you think?

Reply:
That would be aligned with what we do in OpenZeppelin Contracts. Personally I'm ok with that!

Reply:
mmmh having a look at this I believe we can force the definitions to always have a parentheses but for invocations we must leave these alone since we can't differentiate between a modifier invocation and a constructor invocation i.e. `Ownable()`

Reply:
Is it possible to at least remove parentheses when the function is clearly not a constructor? With the consideration that older versions of Solidity used `function ContractName` syntax rather than `constructor`, though I don't know if those are supported by prettier-solidity.

Reply:
Yeah yeah. Been dwelling on the same. We can do that.

Reply:
Ok so this is the new behaviour:
  - ModifierDefinitions with no parameters will always have an empty parentheses.
  - ModifierInvocations with no arguments belonging to a constructor will keep the same parentheses syntax from the input.
  - ModifierInvocations with no arguments not belonging to a constructor will never print parentheses.
I also added a test to compare the `bytecode` to ensure that no harm is done by these changes.

Reply:
Agh, that sucks, but there's no way to avoid that because the parser doesn't have type information. I think that trade-off is ok, but let's not merge this yet because I want to wait for @frangio's input and because I want to add an "edge cases" section to the readme to document this (I'll do that tomorrow).

Reply:
I think this behavior is good.